library/psa_crypto_pake.c

```diff
@@ -238,7 +270,27 @@ psa_status_t mbedtls_psa_pake_setup(mbedtls_psa_pake_operation_t *operation,
             goto error;
         }
 
-        operation->role = role;
+        const size_t user_peer_len = sizeof(jpake_client_id); // client and server have the same length
```

Comment:
That's such a small optimization that I don't know if I wouldn't suggest just having two lengths for the sake of being more readable. If the ID's change then it's a hidden optimization that's not obvious.

Reply:
With two lengths I would have to add two additional conditions to the if statement.